src/Text/Pandoc/Readers/Org.hs

```diff
@@ -407,11 +407,21 @@ orgBlock = try $ do
       _         -> withParsed (fmap $ divWithClass blkType)
 
 blockHeaderStart :: OrgParser (Int, String)
-blockHeaderStart = try $ (,) <$> indent <*> blockType
+blockHeaderStart = try $ (,) <$> spaceLength <*> blockType
  where
-  indent    = length      <$> many spaceChar
   blockType = map toLower <$> (stringAnyCase "#+begin_" *> orgArgWord)
 
+spaceLength :: OrgParser Int
+spaceLength = try $ do
+  tabStop  <- getOption readerTabStop
+  spaceStr <- many spaceChar
+  let charLen = \c ->
+        case c of
+          ' '  -> 1
+          '\t' -> tabStop
+          _    -> 0
+  return $ sum $ map charLen spaceStr
```

Comment:
My personal preference (and, I believe, a widely accepted convention) is to use `.` instead of `$` wherever possible.  So this would become `return . sum $ map charLen spaceStr` or maybe `return . sum . map charLen $ spaceStr`.